mirdata/datasets/irmas.py

```diff
@@ -0,0 +1,251 @@
+
+The IRMAS dataset is offered free of charge for non-commercial use only. You can not redistribute it nor modify it.
+This work is licensed under a Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License
+
+For more details, please visit: https://www.upf.edu/web/mtg/irmas
+
+"""
+
+import os
+import librosa
+
+from mirdata import download_utils
+from mirdata import jams_utils
+from mirdata import core
+from mirdata import utils
+
+BIBTEX = """
+@dataset{juan_j_bosch_2014_1290750,
+  author       = {Juan J. Bosch and Ferdinand Fuhrmann and Perfecto Herrera},
+  title        = {{IRMAS: a dataset for instrument recognition in musical audio signals}},
+  month        = sep,
+  year         = 2014,
+  publisher    = {Zenodo},
+  version      = {1.0},
+  doi          = {10.5281/zenodo.1290750},
+  url          = {https://doi.org/10.5281/zenodo.1290750}
+"""
+
+REMOTES = {
+    'training_data': download_utils.RemoteFileMetadata(
+        filename='IRMAS-TrainingData.zip',
+        url='https://zenodo.org/record/1290750/files/IRMAS-TrainingData.zip?download=1',
+        checksum='4fd9f5ed5a18d8e2687e6360b5f60afe',  # the md5 checksum
+        destination_dir=None,  # relative path for where to unzip the data, or None
+    ),
+    'testing_data_1': download_utils.RemoteFileMetadata(
+        filename='IRMAS-TestingData-Part1.zip',
+        url='https://zenodo.org/record/1290750/files/IRMAS-TestingData-Part1.zip?download=1',
+        checksum='5a2e65520dcedada565dff2050bb2a56',  # the md5 checksum
+        destination_dir=None,  # relative path for where to unzip the data, or None
+    ),
+    'testing_data_2': download_utils.RemoteFileMetadata(
+        filename='IRMAS-TestingData-Part2.zip',
+        url='https://zenodo.org/record/1290750/files/IRMAS-TestingData-Part2.zip?download=1',
+        checksum='afb0c8ea92f34ee653693106be95c895',  # the md5 checksum
+        destination_dir=None,  # relative path for where to unzip the data, or None
+    ),
+    'testing_data_3': download_utils.RemoteFileMetadata(
+        filename='IRMAS-TestingData-Part3.zip',
+        url='https://zenodo.org/record/1290750/files/IRMAS-TestingData-Part3.zip?download=1',
+        checksum='9b3fb2d0c89cdc98037121c25bd5b556',  # the md5 checksum
+        destination_dir=None,  # relative path for where to unzip the data, or None
+    ),
+}
+
+
+DATA = utils.LargeData(
+    'irmas_index.json'
+)
+
+
+INST_DICT = {
+    'cel', 'cla', 'flu', 'gac', 'gel', 'org', 'pia', 'sax', 'tru', 'vio', 'voi'
+}
+
+
+class Track(core.Track):
+    """IRMAS track class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets/Mridangam-Stroke`
+
+    Attributes:
+        track_id (str): track id
+        predominant_instrument (list): Training tracks predominant instrument
```

Comment:
is the description of the different instrument categories documented somewhere?

Reply:
As stated in the comment below, there is just a single instrument category: the predominant instrument. There is no information about the whole instrument line-up of the track, just the predominant one. The important point is that, for the training set there is just a single predominant instrument annotated, while for the testing set there can be more than one predominant instrument in the annotation.
However I was trying to use `instrument` as a catched property to mantain the terminology with the rest of the datasets.

Reply:
Ah i mean the definitions of the three letter instrument codes/ how each instrument category is defined.

Reply:
It's in the attached link, but I will add it to the loader `irmas.py` just in case!

Reply:
Ah yes! It's explained in the documentation on the top of the loader file `irmas.py`. It's also stated in the links attached to the documentation.